crates/cargo-util/src/paths.rs

```diff
@@ -539,7 +539,18 @@ fn _link_or_copy(src: &Path, dst: &Path) -> Result<()> {
         // gory details.
         fs::copy(src, dst).map(|_| ())
     } else {
-        fs::hard_link(src, dst)
+        if cfg!(target_os = "macos") {
+            // This is a work-around for a bug on macos with Apple Silicon
+            // Since Apple Silicon processor, all binary required to be signed.
```

Comment:
I think this comment needs to be updated.

Reply:
Updated. As we don't have a definite answer on what the problem is. It is hard to put proper explanations in the comments. We only know that using clone instead of hardlink works.